receiver/hostmetricsreceiver/internal/scraper/memoryscraper/memory_scraper_test.go

```diff
@@ -16,61 +16,77 @@ package memoryscraper
-		// expect 1 metric
-		assert.Equal(t, 1, metrics.Len())
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			scraper := newMemoryScraper(context.Background(), &Config{})
+			if test.virtualMemoryFunc != nil {
+				scraper.virtualMemory = test.virtualMemoryFunc
+			}
 
-		// for memory used metric, expect a datapoint for all 6 state labels
-		memoryUsageMetric := metrics.At(0)
-		internal.AssertDescriptorEqual(t, memoryUsageDescriptor, memoryUsageMetric.MetricDescriptor())
-		assert.Equal(t, 6, memoryUsageMetric.Int64DataPoints().Len())
-		internal.AssertInt64MetricLabelHasValue(t, memoryUsageMetric, 0, stateLabelName, usedStateLabelValue)
-		internal.AssertInt64MetricLabelHasValue(t, memoryUsageMetric, 1, stateLabelName, freeStateLabelValue)
-		internal.AssertInt64MetricLabelHasValue(t, memoryUsageMetric, 2, stateLabelName, bufferedStateLabelValue)
-		internal.AssertInt64MetricLabelHasValue(t, memoryUsageMetric, 3, stateLabelName, cachedStateLabelValue)
-		internal.AssertInt64MetricLabelHasValue(t, memoryUsageMetric, 4, stateLabelName, slabReclaimableStateLabelValue)
-		internal.AssertInt64MetricLabelHasValue(t, memoryUsageMetric, 5, stateLabelName, slabUnreclaimableStateLabelValue)
-	})
-}
+			err := scraper.Initialize(context.Background())
+			require.NoError(t, err, "Failed to initialize memory scraper: %v", err)
+			defer func() { assert.NoError(t, scraper.Close(context.Background())) }()
+
+			metrics, err := scraper.ScrapeMetrics(context.Background())
+			if test.expectedErr != "" {
+				assert.EqualError(t, err, test.expectedErr)
+				return
+			}
+			require.NoError(t, err, "Failed to scrape metrics: %v", err)
 
-func createScraperAndValidateScrapedMetrics(t *testing.T, config *Config, assertFn validationFn) {
-	scraper := newMemoryScraper(context.Background(), config)
-	err := scraper.Initialize(context.Background())
-	require.NoError(t, err, "Failed to initialize memory scraper: %v", err)
-	defer func() { assert.NoError(t, scraper.Close(context.Background())) }()
+			assert.Equal(t, 1, metrics.Len())
 
-	metrics, err := scraper.ScrapeMetrics(context.Background())
-	require.NoError(t, err, "Failed to scrape metrics: %v", err)
+			assertMemoryUsageMetricValid(t, metrics.At(0), memoryUsageDescriptor)
+
+			if runtime.GOOS == "linux" {
+				assertMemoryUsageMetricHasLinuxSpecificStateLabels(t, metrics.At(0))
+			} else if runtime.GOOS != "windows" {
+				internal.AssertInt64MetricLabelHasValue(t, metrics.At(0), 2, stateLabelName, inactiveStateLabelValue)
+			}
+		})
+	}
+}
+
+func assertMemoryUsageMetricValid(t *testing.T, metric pdata.Metric, descriptor pdata.MetricDescriptor) {
+	internal.AssertDescriptorEqual(t, descriptor, metric.MetricDescriptor())
+	assert.GreaterOrEqual(t, metric.Int64DataPoints().Len(), 2)
+	internal.AssertInt64MetricLabelHasValue(t, metric, 0, stateLabelName, usedStateLabelValue)
+	internal.AssertInt64MetricLabelHasValue(t, metric, 1, stateLabelName, freeStateLabelValue)
```

Comment:
This test will not pass for non windows non Linux systems correct?

Reply:
Should be fine - "used" & "free" state labels are returned for all OSs

Reply:
Confirmed test passing on Mac

Reply:
I missed used on the other, saw only free and inactive